### PR TITLE
Fix/docs

### PR DESCRIPTION
**Pull request by cqc-melf**
try to clean up everything in the repo, only keeping the docs and the readme. Is there anything else we should keep at the moment?

### CODE REVIEW

**Comment by cqc-melf** [comment]
Any objections on removing the required checks and removing all secrets? @cqc-alec

**Comment by cqc-alec** [comment]
> Any objections on removing the required checks and removing all secrets? @cqc-alec

No objections.

**Comment by cqc-melf** [comment]
> > Any objections on removing the required checks and removing all secrets? @cqc-alec
> 
> No objections.

done

File: generate_mgit.sh [hunk 1]
```diff
@@ -0,0 +1,84 @@
+#!/bin/bash
```
Comment (cqc-melf) [inline generate_mgit.sh:1]: not sure if this is something other people would use and not sure if this is the right place to have it.

File: .github/workflows/docs/build-docs [hunk 2]
```diff
@@ -59,16 +71,24 @@ def build_module_docs(module):
         content = f.readlines()
         content.append("\n.. toctree::\n\t:caption: Extensions:\n\t:maxdepth: 1\n\n")
         content.append(f"\tIntroduction <../index.html#http://>\n")
-        for mod in get_all_modules():
```
Comment (cqc-alec) [inline .github/workflows/docs/build-docs:62]: I guess `get_all_modules()` and maybe some other functions can be deleted?

File: generate_mgit.sh (hunk 1 above)
Comment (cqc-alec) [reply generate_mgit.sh:1]: Seems as good a place as any, and could be useful to others! But maybe put it inside a directory called e.g. `dev-utils`?

**Review by cqc-alec — CHANGES_REQUESTED** [review]

File: generate_mgit.sh (hunk 1 above)
Comment (cqc-melf) [reply generate_mgit.sh:1]: done

File: .github/workflows/docs/build-docs (hunk 2 above)
Comment (cqc-melf) [reply .github/workflows/docs/build-docs:62]: I have done some clean up work in that file, removed this and other things.

**Review by cqc-alec — APPROVED** [review]

**Pull request merged**